src/app/pages/MarginTradePage/components/TradeDialog/index.tsx

```diff
@@ -27,29 +32,60 @@ import {
 import { TxDialog } from '../../../../components/Dialogs/TxDialog';
 import { LoadableValue } from '../../../../components/LoadableValue';
 import { Dialog } from '../../../../containers/Dialog';
-import { PricePrediction } from '../../../../containers/MarginTradeForm/PricePrediction';
 import { useApproveAndTrade } from '../../../../hooks/trading/useApproveAndTrade';
 import { useTrading_resolvePairTokens } from '../../../../hooks/trading/useTrading_resolvePairTokens';
 import { useAccount } from '../../../../hooks/useAccount';
+
+import { useCookie } from 'app/hooks/useCookie';
+import { useAffiliates_getAffiliatesUserReferrer } from 'app/hooks/affiliates/useAffiliates_getAffiliatesUserReferrer';
 import { selectMarginTradePage } from '../../selectors';
 import { actions } from '../../slice';
 import { LiquidationPrice } from '../LiquidationPrice';
 import { TxFeeCalculator } from '../TxFeeCalculator';
 import { TradingPosition } from 'types/trading-position';
+import { useGetEstimatedMarginDetails } from '../../../../hooks/trading/useGetEstimatedMarginDetails';
+import { useCurrentPositionPrice } from '../../../../hooks/trading/useCurrentPositionPrice';
 
 const maintenanceMargin = 15000000000000000000;
 
 export function TradeDialog() {
+  const [referrer, setReferrer] = useState<string | null>(null);
   const { t } = useTranslation();
   const account = useAccount();
+  const { get: getCookie } = useCookie();
   const { checkMaintenance, States } = useMaintenance();
   const openTradesLocked = checkMaintenance(States.OPEN_MARGIN_TRADES);
   const { position, amount, pairType, collateral, leverage } = useSelector(
     selectMarginTradePage,
   );
-  // const [slippage, setSlippage] = useState(0.5);
+  const [slippage, setSlippage] = useState(0.5);
   const dispatch = useDispatch();
 
+  const {
+    value,
+    loading: referrerLoading,
+    error,
+  } = useAffiliates_getAffiliatesUserReferrer(account);
+
+  useEffect(() => {
+    if (!referrerLoading) {
+      if (error) {
+        console.log(error, value);
+        return;
+      }
+      const referralWallet =
+        value !== ethGenesisAddress ? value : getCookie(sovAffiliateCookie);
+      if (
+        referralWallet?.indexOf('0x') === 0 &&
+        referralWallet?.length === 42 &&
```

Comment:
Extract this condition and condition before to a util method `isRskAddressFormat`. Otherwise, it just looks like a bunch of magic numbers and strings.